### PR TITLE
Additional Catch2 tests of existing functionality.

### PR DESCRIPTION
**Pull request by jdeaton**
This PR is an extension of #60 and adds additional tests of existing MAST library capability. Most tests were originally implemented by @JohnDN90.

Below is a checklist of commits implemented by @JohnDN90 in his `add_catch2_tests` branch the whose contents/functionality need rolled in:
- [x] Added catch2 header file for upcoming new tests. (#60)
- [x] Bug fixes and enhancements to CMake files. (#60)
- [x] Added option to use libmesh_devel (development) version. (#60)
- [X] Added catch2 tests for some MAST base functions. (#60)
- [x] Added catch2 test for TEMPERATURE boundary condition.
- [x] Added catch2 tests for mast_boundary_condition_base
- [x] Removed shear coefficient (kappa) from isotropic material. Added plane strain support to isotropic material. (#60)
- [x] Added catch2 tests for mast_isotropic_material. (#60)
- [x] Multiple changes to orthotropic material, including added plane strain support. (#60)
- [x] Added catch2 tests for orthotropic_material_property_card. (#60)
- [x] Moved shear coefficient (kappa) to 2D section card and added overloaded functions to 2D section card. (#60)
- [x] Added catch2 tests for solid_2d_section_element_property_card. (this PR)
- [x] Added overloaded functions to isotropic 3D section property card. (this PR)
- [x] Added catch2 tests for isotropic_element_property_card_3D. (this PR)
- Added new public release statement to README.
- [x] Add overloaded functions to solid_1d_section_element_property_card. (this PR)
- Formatting for README.md
- [x] Multiple changes to 1d_solid_section_element_property_card. (this PR)
- [x] Added catch2 tests for solid_1d_section_element_property_card. (this PR)
- Added basic catch2 tests for generic 1D structural elements.
- Fixed bug in GitHub Issue #41 where NO_BENDING incorrectly called TIMOSHENKO.
- Added catch2 tests for 1D structural element with extension and torsion.
- [x] Fixed bug in GitHub issue #43, where incorrect I value was being used for bending about y-axis. (#75)
- Added many for tests for 1D structural 2-noded (edge2) element.
- Added generic catch2 tests for 2D structrual elements.
- Added many catch2 tests for 4-node quadrilaterial 2D element (quad4).
- [X] UPDATE EXAMPLES MOVING KAPPA FROM MATERIAL TO SECTION CARD (#60)
- [X] Fixed bugs in catch2 tests which caused some to fail when run with a debug version of MAST. (some fixed in this PR)

### CODE REVIEW

File: tests/property/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+        FIXTURES_REQUIRED Isotropic_Material_2D_Thermoelastic_mpi
+        FIXTURES_SETUP Element_Property_Card_2D_Thermoelastic_mpi)
+
+
+# Dynamic
+add_test(NAME Element_Property_Card_2D_Dynamic
+    COMMAND $<TARGET_FILE:mast_catch_tests> -w NoTests "element_property_card_constant_dynamic_2d")
+set_tests_properties(Element_Property_Card_2D_Dynamic
+    PROPERTIES
+        LABELS "SEQ"
```
Comment (jdeaton) [inline tests/property/CMakeLists.txt]: @JohnDN90 based on the pattern of the other tests, are we missing a `FIXTURES_REQUIRED` here?

File: tests/property/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+        FIXTURES_SETUP Element_Property_Card_2D_Dynamic)
+
+add_test(NAME Element_Property_Card_2D_Dynamic_mpi
+    COMMAND ${MPIEXEC_EXECUTABLE} -np 2 $<TARGET_FILE:mast_catch_tests> -w NoTests "element_property_card_constant_dynamic_2d")
+set_tests_properties(Element_Property_Card_2D_Dynamic_mpi
+    PROPERTIES
+        LABELS "MPI"
```
Comment (jdeaton) [inline tests/property/CMakeLists.txt]: @JohnDN90 same as above?

**Comment by jdeaton** [comment]
This PR has a good chunk of changes in it and completes the addition of tests for a lot of fundamental MAST infrastructure. The remaining commits begin to integrate tests of actual elements/Jacobians/etc. I think I will split this work and merge this in once CI clears and continue the test integration in another PR.
